schematic/store/synapse.py

```diff
@@ -680,105 +704,58 @@ def getFilesInStorageDataset(
         Raises:
             ValueError: Dataset ID not found.
         """
-        # select all files within a given storage dataset folder (top level folder in
-        # a Synapse storage project or folder marked with contentType = 'dataset')
-        walked_path = synapseutils.walk(
-            self.syn, datasetId, includeTypes=["folder", "file"]
-        )
-
-        current_entity_location = self.synapse_entity_tracker.get(
-            synapse_id=datasetId, syn=self.syn, download_file=False
-        )
-
-        def walk_back_to_project(
-            current_location: Entity, location_prefix: str, skip_entry: bool
-        ) -> str:
-            """
-            Recursively walk back up the project structure to get the paths of the
-            names of each of the directories where we started the walk function.
-
-            Args:
-                current_location (Entity): The current entity location in the project structure.
-                location_prefix (str): The prefix to prepend to the path.
-                skip_entry (bool): Whether to skip the current entry in the path. When
-                    this is True it means we are looking at our starting point. If our
-                    starting point is the project itself we can go ahead and return
-                    back the project as the prefix.
-
-            Returns:
-                str: The path of the names of each of the directories up to the project root.
-            """
-            if (
-                skip_entry
-                and "concreteType" in current_location
-                and current_location["concreteType"] == PROJECT_ENTITY
-            ):
-                return f"{current_location.name}/{location_prefix}"
+        file_list = []
 
-            updated_prefix = (
-                location_prefix
-                if skip_entry
-                else f"{current_location.name}/{location_prefix}"
-            )
-            if (
-                "concreteType" in current_location
-                and current_location["concreteType"] == PROJECT_ENTITY
-            ):
-                return updated_prefix
-            current_location = self.synapse_entity_tracker.get(
-                synapse_id=current_location["parentId"],
-                syn=self.syn,
-                download_file=False,
+        if self.storageFileviewTable.empty:
```

Comment:
@GiaJordan thanks for adding this, but lets take a step back and think about where this should be added?  Should this be added here or within `synapse.py`?

Reply:
> Should this be added here or within synapse.py?
I can add it to a different part of the `SynapseStorage` class in `synapse.py` instead? Like after the query is performed, unless you meant a different file?  @thomasyu888

Reply:
@thomasyu888 I originally put it there because I didn't want to prohibit any functionality that could be valid with an empty file view. For instance later in this method the case where someone was trying to generate a manifest for a new dataset without files and no manifest already existing. 
The intention for `query_fileview` was that we could use it throughout schematic to query only the parts of the view we need for the operation initially to reduce the amount of subsequent queries we'd need to make (there wasn't time to do this kind of scoping within this ticket) but I thought that as we moved away from just taking everything in the view there would be more cases where an empty view is a valid result and tells us something useful for an operation as well.
I don't recall ever having issues with completely empty views, possibly because how users have to manually set those up.
Just a note that I see these cases as separate from when a view is queried after files have been recently modified but the view hasn't been updated; we're using [tenacity](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/store/synapse.py#L2883) to cover that case.

Reply:
Thanks @GiaJordan for the explanation and context.  I think we can keep this here as is.  The chances of having an empty view is probably low in the grand scheme of things anyhow. 

I would add a unit test to make sure this exception is caught, but after reviewal of https://github.com/Sage-Bionetworks/schematic/pull/1554 since there's some tests implemented in this PR that won't pass due to the HACK i implemented there.

Reply:
Thanks Tom and Gianna. Adding the code here to deal with empty file view makes sense to me.